apps/wallet/src/ui/app/components/known-scam-overlay/useDappPreflight.ts

```diff
@@ -1,23 +1,31 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
+import { type API_ENV } from '_src/shared/api-env';
 import { useAppsBackend } from '@mysten/core';
 import { useSuiClient } from '@mysten/dapp-kit';
 import { type Transaction } from '@mysten/sui/transactions';
 import { toB64 } from '@mysten/sui/utils';
 import { useQuery } from '@tanstack/react-query';
 
-import { RequestType, type DappPreflightRequest, type DappPreflightResponse } from './types';
+import {
+	API_ENV_TO_NETWORK,
+	RequestType,
+	type DappPreflightRequest,
+	type DappPreflightResponse,
+} from './types';
 
 export function useDappPreflight({
 	requestType,
 	origin,
 	transaction,
 	requestId,
+	network,
 }: {
 	requestType: RequestType;
 	origin?: string;
 	transaction?: Transaction;
 	requestId: string;
+	network: API_ENV;
```

Comment:
any reason to not query for apiEnv directly here rather than using a prop?
const { apiEnv } = useAppSelector((state) => state.app);

Reply:
doesn't matter too much since these hooks should probably be consolidated anyway, but the original intent was that one hook is responsible for the request and the other is responsible for the logic around showing the warning